scripts/posh-vcpkg.psm1

```diff
@@ -1,39 +1,41 @@
-param()
+Register-ArgumentCompleter -Native -CommandName vcpkg -ScriptBlock {
+    param(
+        [string]$wordToComplete,
+        [System.Management.Automation.Language.CommandAst]$commandAst,
+        [int]$cursorPosition
+    )
 
-if (Get-Module posh-vcpkg) { return }
+    if ($cursorPosition -lt $commandAst.CommandElements[0].Extent.EndOffset) {
+        return
+    }
 
-if ($PSVersionTable.PSVersion.Major -lt 5) {
-    Write-Warning ("posh-vcpkg does not support PowerShell versions before 5.0.")
-    return
-}
+    [string]$commandText = $commandAst.CommandElements[0].Value
 
-if (Test-Path Function:\TabExpansion) {
-    Rename-Item Function:\TabExpansion VcpkgTabExpansionBackup
-}
+    [string[]]$textsBeforeCursor = $commandAst.CommandElements |
+        Select-Object -Skip 1 | ForEach-Object {
+            if ($_.Extent.EndOffset -le $cursorPosition) {
+                $_.Extent.Text
+            }
+            elseif ($_.Extent.StartOffset -lt $cursorPosition) {
+                $_.Extent.Text.Substring(0, $cursorPosition - $_.Extent.StartOffset)
+            }
+        }
 
-function TabExpansion($line, $lastWord) {
-    $lastBlock = [regex]::Split($line, '[|;]')[-1].TrimStart()
+    $spaceToComplete = if ($wordToComplete -ne '') { $null }
+    elseif ($PSNativeCommandArgumentPassing -in 'Standard', 'Windows') { '' }
+    else { '""' }
 
-    switch -regex ($lastBlock) {
-        "^(?<vcpkgexe>(\./|\.\\|)vcpkg(\.exe|)) (?<remaining>.*)$"
-        {
-            & $matches['vcpkgexe'] autocomplete $matches['remaining']
-            return
-        }
+    [PowerShell]$cmd = [PowerShell]::Create().AddScript{
+        Set-Location $args[0]
+        & $args[1] autocomplete @($args[2])
+    }.AddParameters(($PWD, $commandText, @($textsBeforeCursor + $spaceToComplete)))
 
-        # Fall back on existing tab expansion
-        default {
-            if (Test-Path Function:\VcpkgTabExpansionBackup) {
-                VcpkgTabExpansionBackup $line $lastWord
-            }
-        }
-    }
-}
+    [string[]]$completions = $cmd.Invoke()
 
-$exportModuleMemberParams = @{
-    Function = @(
-        'TabExpansion'
-    )
+    if ($cmd.HadErrors -or $completions.Count -eq 0) {
+        return

```

Comment:
e.g. `vcpkg install versions^` (No port named `versions` but directory `.\versions\`)

Unable to prevent fallback to default completion.

(Unless get an error, like return empty string.)

It will fallback default with:

* `return`
* `return $null`
* `return @()`
* `return [System.Collections.Generic.List[System.Management.Automation.CompletionResult]]::new()`

See https://github.com/PowerShell/PowerShell/issues/19628